lib/types.nix

```diff
@@ -470,6 +470,18 @@ rec {
       nestedTypes.elemType = elemType;
     };
 
+    unique = { message }: type: mkOptionType rec {
```

Comment:
Same question: shouldn't this replace `uniq`?

Reply:
That would be an unnecessary breaking change for out-of-tree usage of the module system. `uniq` is an awful name anyway, so I'd rather just deprecate that at some point.

Reply:
I meant something like `uniq = unique { message = "" };`, maybe with a deprecation warning. Sorry, that wasn't clear

Reply:
I think it's ok not to deprecate or not deprecate yet.